core/util.go

```diff
@@ -481,61 +483,96 @@ type fileOrDirectory interface {
 	Evaluatable
 }
 
-// execInMount is a helper used by Directory.execInMount and File.execInMount
+// execInMount evaluates a file or directory, mounts it, then calls the supplied callback function.
 func execInMount[T fileOrDirectory](ctx context.Context, obj T, f func(string) error, optFns ...execInMountOptFn) (T, error) {
+	root, closer, err := execInMountCloser(ctx, obj, optFns...)
+	if err != nil {
+		return nil, err
+	}
+	err = f(root)
+	if err != nil {
+		_, _ = closer(true)
+		return nil, err
+	}
+	return closer(false)
+}
+
+// execInMountCloser evaluates an object and mounts the root fs and returns the mounted path and a closer, which will unmount
+// the file or directory object's root filesystem, and potentially return a modified object, if both the withSavedSnapshot option is specified and the abort flag was not set.
+// The abort flag is only used when the withSavedSnapshot option is specified.
+// NOTE: prefer execInMount where possible, unless finer-grained control of the filesystem mount is required.
+func execInMountCloser[T fileOrDirectory](ctx context.Context, obj T, optFns ...execInMountOptFn) (string, func(abort bool) (T, error), error) {
```

Comment:
Could readonly be a `execInMountOptFn`? To avoid it needing to propagate everywhere.

Reply:
I removed the `ReadonlyMountRef` function, and added a `mountRefAsReadOnly` option.

The `mountObj` function (fka `execInMountCloser`), is different enough from `MountRef` (e.g. it also performs the snapshot call), so I decided to keep the opts structs separate. I did however plumb through the case where `!opt.commitSnapshot` will result in the readonly flag being set.